dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/DatasetReportService.java

```diff
@@ -0,0 +1,319 @@
+        for (FileMetadata fileMetadata : allFilesMetadataSorted) {
+            processFile(fileMetadata, csvPrinter, datasetVersionRecord);
+        }
+    }
+
+    private Record createDatasetVersionRecord(DatasetVersion datasetVersion, Record datasetRecord) {
+        Record datasetVersionRecord = new Record(datasetRecord);
+        datasetVersionRecord.setDeaccessionData(datasetVersion.isDeaccessioned() ? datasetVersion.getVersionNote() : StringUtils.EMPTY);
+        DatasetField depositDate = datasetVersion.getDatasetFieldByTypeName(DatasetFieldConstant.dateOfDeposit).orElse(null);
+        datasetVersionRecord.setDepositDate(depositDate != null ? depositDate.getValue() : StringUtils.EMPTY);
+        datasetVersionRecord.setVersionNumber(datasetVersion.getFriendlyVersionNumber());
+        datasetVersionRecord.setDatasetVersionPublicationDate(datasetVersion.getPublicationDateAsString());
+        datasetVersionRecord.setDatasetVersionState(Objects.toString(datasetVersion.getVersionState(), StringUtils.EMPTY));
+        return datasetVersionRecord;
+    }
+
+    private void processFile(FileMetadata fileMetadata, CSVPrinter csvPrinter, Record datasetVersionRecord) throws IOException {
+        DataFile dataFile = fileMetadata.getDataFile();
+        Record fileRecord = createFileRecord(datasetVersionRecord, dataFile);
+        csvPrinter.printRecord(fileRecord.getValues());
+    }
+
+    private Record createFileRecord(Record datasetVersionRecord, DataFile dataFile) {
+        Record fileRecord = new Record(datasetVersionRecord);
+        fileRecord.setFileName(dataFile.getDisplayName());
+        fileRecord.setFileId(dataFile.getId());
+        fileRecord.setChecksum(dataFile.getChecksumValue());
+        fileRecord.setChecksumType(Objects.toString(dataFile.getChecksumType(), StringUtils.EMPTY));
+        fileRecord.setSize(dataFile.getFilesize());
+        Long uncompressedSize = dataFile.getUncompressedSize();
+        fileRecord.setSizeDecompressed(uncompressedSize != 0L ? uncompressedSize : null);
+        String tags = dataFile.getTags().stream()
+                .map(DataFileTag::getTypeLabel)
+                .collect(Collectors.joining(" "));
+        fileRecord.setTags(tags);
+        fileRecord.setDatafilePublicationDate(dataFile.getPublicationDateFormattedYYYYMMDD());
+        String license = Optional.ofNullable(dataFile.getFileMetadata())
+                .map(FileMetadata::getTermsOfUse)
+                .map(FileTermsOfUse::getLicense)
+                .map(License::getName).orElse(StringUtils.EMPTY);
+        fileRecord.setLicense(license);
+        fileRecord.setContentType(dataFile.getContentType());
+        fileRecord.setNumberOfDownloads(guestbookResponseService.getCountGuestbookResponsesByDataFileId(dataFile.getId()));
+        return fileRecord;
+    }
+
+    /**
+     * This enum establishes the columns of CSV report and their order
+     */
+    private enum FileDataField {
```

Comment:
Did you consider passing this as a parameter rather than 'hard code' it here? That would allow creation of other reports based on this one - like simpler report with only few fields.

Reply:
I think I'll change visibility of this enum to package-private so it could be reached by other classes (also in integration test). I think that currently there is no need to extract this class to upper level as we're not going to create more reports now, and if so then it will be a straightforward task.

Reply:
I've deleted previous comment.

Reply:
I'm not sure whether I understand correctly your idea.

Reply:
I haven't thought that thru :) but I'd imagine being able to reuse `createReport` logic just with a subset of current columns or with another order. 
Since creating more reports is doubtful right now that's probably an overkill for this task. I only wanted to ask if you considered more flexible approach.

Reply:
Ok, now I understand. It is probably possible – we could overload `getValues()` and `getHeaders()` methods to take the list of enum values and to return only that fields. But as for now it seems unnecessary to add that logic.